MagicBots/Overseer/Framework/Discord/OverseerHostedService.cs

```diff
@@ -1,23 +1,38 @@
 ﻿using Discord;
 using Discord.WebSocket;
+using MagicBots.Overseer.Framework.Services;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
 
 namespace MagicBots.Overseer.Framework.Discord
 {
     public class OverseerHostedService : DiscordClientHostedService
     {
-        public OverseerHostedService(ILogger logger, IConfiguration configuration) : base(logger, configuration)
+        private readonly TriggeringService _triggeringService;
+
+        public OverseerHostedService(ILogger logger, IConfiguration configuration,
+            TriggeringService triggeringService) : base(logger, configuration)
         {
+            _triggeringService = triggeringService;
+
+            Client!.MessageReceived += MessageReceivedAsync;
         }
 
         protected override string ConfigSectionName => "DiscordOverseer";
 
+        private async Task MessageReceivedAsync(SocketMessage rawMessage)
+        {
+            // Ignore system messages, or messages from other bots
+            if (!(rawMessage is SocketUserMessage message))
+                return;
+            if (message.Source != MessageSource.User)
+                return;
+
+            var context = new DiscordTriggerContext(Client!, message);
+            await _triggeringService.ProcessMessage(context);
```

Comment:
Is there any deadlock potential here? I think discord generally recommends not running any long-running tasks in the event handlers. Instead you would normally kick off a new async task that's not awaited

Reply:
I suppose you could also leave that up to the actual module that uses the trigger

Reply:
The triggering system actually does spin off a new async task. I added some Todos to investigate this more in the future.

Reply:
Ok perfect, I must have just not noticed that